tests/Framework.py

```diff
@@ -468,13 +486,15 @@ def setUp(self):
         github.Requester.Requester.setDebugFlag(True)
         github.Requester.Requester.setOnCheckMe(self.getFrameChecker())
 
-        self.g = self.get_github(self.retry, self.pool_size)
+        self.g = self.get_github(self.authMode, self.retry, self.pool_size)
 
-    def get_github(self, retry, pool_size):
-        if self.authMode == "token":
+    def get_github(self, authMode, retry=None, pool_size=None):
+        if authMode == "token":
             auth = self.oauth_token
-        elif self.authMode == "jwt":
+        elif authMode == "jwt":
             auth = self.jwt
+        elif authMode == "app":
+            auth = self.app_auth
         else:
             raise ValueError(f"Unsupported test auth mode: {self.authMode}")
```

Comment:
>_possible bug_

**✨ AI Suggestion:**The ValueError message uses `self.authMode` instead of the `authMode` parameter. This could lead to confusing error messages if an invalid mode is passed directly to `get_github`, as the message would report the instance's default mode rather than the invalid one provided.
```suggestion
        else:
            raise ValueError(f"Unsupported test auth mode: {authMode}")
```